PARTIE 1: IMPORT DES LIBRAIRIES USUELLES

In [79]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

PARTIE 2: EXPLORATION DU DATASET

In [80]:
df=pd.read_csv("Speed+Dating+Data.csv", encoding='latin-1')
df.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 195 columns]

In [81]:
df["race"].sort_values()

4421    1.0
7359    1.0
7358    1.0
7357    1.0
7356    1.0
       ... 
5127    NaN
5128    NaN
5129    NaN
5130    NaN
5131    NaN
Name: race, Length: 8378, dtype: float64

In [82]:
df.describe(include="all")

iid           id       gender          idg       condtn  \
count   8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     283.675937     8.960248     0.500597    17.327166     1.828837   
std      158.583367     5.491329     0.500029    10.940735     0.376673   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      154.000000     4.000000     0.000000     8.000000     2.000000   
50%      281.000000     8.000000     1.000000    16.000000     2.000000   
75%      407.000000    13.000000     1.000000    26.000000     2.000000   
max      552.000000    22.000000     1.000000    44.000000     2.000000   

               wave        round     position     positin1        order  ...  \
count   8378.000000  8378.000000  8378.000000  6532.000000  8378.000000  ...   
unique          NaN          NaN          NaN          NaN          NaN  ...   
top             NaN          NaN          NaN          NaN          NaN  ...   
freq            NaN          NaN          NaN          NaN          NaN  ...   
mean      11.350919    16.872046     9.042731     9.295775     8.927668  ...   
std        5.995903     4.358458     5.514939     5.650199     5.477009  ...   
min        1.000000     5.000000     1.000000     1.000000     1.000000  ...   
25%        7.000000    14.000000     4.000000     4.000000     4.000000  ...   
50%       11.000000    18.000000     8.000000     9.000000     8.000000  ...   
75%       15.000000    20.000000    13.000000    14.000000    13.000000  ...   
max       21.000000    22.000000    22.000000    22.000000    22.000000  ...   

            attr3_3      sinc3_3     intel3_3       fun3_3       amb3_3  \
count   3974.000000  3974.000000  3974.000000  3974.000000  3974.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       7.240312     8.093357     8.388777     7.658782     7.391545   
std        1.576596     1.610309     1.459094     1.744670     1.961417   
min        2.000000     2.000000     3.000000     2.000000     1.000000   
25%        7.000000     7.000000     8.000000     7.000000     6.000000   
50%        7.000000     8.000000     8.000000     8.000000     8.000000   
75%        8.000000     9.000000     9.000000     9.000000     9.000000   
max       12.000000    12.000000    12.000000    12.000000    12.000000   

            attr5_3      sinc5_3     intel5_3       fun5_3       amb5_3  
count   2016.000000  2016.000000  2016.000000  2016.000000  2016.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean       6.810020     7.615079     7.932540     7.155258     7.048611  
std        1.507341     1.504551     1.340868     1.672787     1.717988  
min        2.000000     2.000000     4.000000     1.000000     1.000000  
25%        6.000000     7.000000     7.000000     6.000000     6.000000  
50%        7.000000     8.000000     8.000000     7.000000     7.000000  
75%        8.000000     9.000000     9.000000     8.000000     8.000000  
max       10.000000    10.000000    10.000000    10.000000    10.000000  

[11 rows x 195 columns]

In [83]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8378 entries, 0 to 8377
Columns: 195 entries, iid to amb5_3
dtypes: float64(174), int64(13), object(8)
memory usage: 12.5+ MB


In [84]:
df.shape

(8378, 195)

VISUALISATIONS:

In [85]:
# Visualisation de la distribution des participants par genre et par wave

fig=px.histogram(df, x='wave', color='gender', barmode='group', title="Distribution des participants par genre et par wave")
fig.show()

D'après la visualisation ci-dessus, nous remarquons qu'il y'a autant de femmes que d'hommes.

In [86]:
# Visualisation de la distribution de l'âge des participants

fig = px.box(df, y='age', title="Distribution des âges des participants")
fig.update_layout(width=500, height=500)
fig.show()

Nous remarquons alors que l'âge des participants se situe plutôt entre 24 et 28 ans.

In [87]:
# Visualisation de la distribution de la race

df["race"] = df["race"].apply(lambda x: 'Black/African American' if x == 1
                                  else 'European/Caucasian-American' if x == 2
                                  else 'Latino/Hispanic American' if x == 3
                                  else 'Asian/Pacific Islander/Asian-Americann' if x == 4
                                  else 'Native American' if x == 5
                                   else 'Other'
                                         )


In [88]:
# Visualisation de la distribution de la race
df["race"].value_counts()

race
European/Caucasian-American               4727
Asian/Pacific Islander/Asian-Americann    1982
Latino/Hispanic American                   664
Other                                      585
Black/African American                     420
Name: count, dtype: int64

In [89]:
# Visualisation de la distribution de la race

color_palette = px.colors.sequential.Reds[::-1] 

race = df["race"].value_counts(ascending=False)

trace = go.Pie(labels=race.index, values=race, marker=dict(colors=color_palette))

layout = go.Layout(
    title={'text': 'Distribution of the race of the participants', 'x': 0.5,'yanchor': 'top', 'y': 0.97},
    width=700, 
    height=500,
        legend=dict(
        x=0.9,
        y=0.3   #Pour définir la position de la légende
    )
    )
fig = go.Figure(data=[trace], layout=layout)
fig.show()

Dans ce qui suit, nous allons prendre que les "waves" de 1,2,3,4 et 5 et celles qui sont supérieures à 10.

QUESTION1: LES ATTRIBUTS LES MOINS DESIRABLES

In [90]:
# Avant le speed date

colonnes_avant_speed_date=['attr1_1','sinc1_1','intel1_1','fun1_1','amb1_1','shar1_1']

    # Pour les femmes 
df_femmes_waves = df[(df['gender'] == 0) & ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question11_femme = df_femmes_waves[colonnes_avant_speed_date]

    # Pour les hommes

df_hommes_waves = df[(df['gender'] == 1) & ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question11_homme = df_hommes_waves[colonnes_avant_speed_date]

# Après le speed date
    # Pour les femmes
        # Si les femmes matchent

df_femmes_waves_match1=df[(df['gender'] == 0) & (df['match']==1) & ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question121_femme = df_femmes_waves_match1[colonnes_avant_speed_date]

        # Si les femmes ne matchent pas

df_femmes_waves_match0=df[(df['gender'] == 0) & (df['match']==0)& ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question120_femme = df_femmes_waves_match0[colonnes_avant_speed_date]

    # Pour les hommes
        # Si les hommes matchent

df_hommes_waves_match1 = df[(df['gender'] == 1) & (df['match']==1) & ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question121_homme = df_hommes_waves_match1[colonnes_avant_speed_date]

        # Si les hommes ne matchent pas

df_hommes_waves_match0 = df[(df['gender'] == 1) & (df['match']==0) & ((df['wave'] <= 5) | (df['wave'] >= 10))]
df_question120_homme = df_hommes_waves_match0[colonnes_avant_speed_date]

In [91]:
# Avant le speed date
df_question11_femme_mean=df_question11_femme.mean()
df_question11_homme_mean=df_question11_homme.mean()
data1 = np.vstack([df_question11_femme_mean, df_question11_homme_mean]) #np.vstack est utilisé pour empiler les moyennes en verticale
df_question11_global=pd.DataFrame(data1, columns=colonnes_avant_speed_date,index=['Femmes', 'Hommes'])
print("La moyenne des valeurs des attributs selon femmes et hommes avant le speed date:")
display(df_question11_global)

# Si les femmes/hommes matchent
df_question121_femme_mean=df_question121_femme.mean()
df_question121_homme_mean=df_question121_homme.mean()
data2 = np.vstack([df_question121_femme_mean, df_question121_homme_mean]) 
df_question121_global=pd.DataFrame(data2, columns=colonnes_avant_speed_date, index=['Femmes', 'Hommes'])
print("La moyenne des valeurs des attributs selon femmes et hommes s'ils matchent avec le partenaire:")
display(df_question121_global)

# Si les femmes/hommes ne matchent pas
df_question120_femme_mean=df_question120_femme.mean()
df_question120_homme_mean=df_question120_homme.mean()
data3 = np.vstack([df_question120_femme_mean, df_question120_homme_mean])
df_question120_global=pd.DataFrame(data3, columns=colonnes_avant_speed_date, index=['Femmes', 'Hommes'])
print("La moyenne des valeurs des attributs selon femmes et hommes s'ils ne matchent pas avec le partenaire:")
display(df_question120_global)



La moyenne des valeurs des attributs selon femmes et hommes avant le speed date:


attr1_1    sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1
Femmes  18.788883  18.377089  21.568459  17.043556  12.005635  12.264193
Hommes  29.108895  16.233127  19.564615  17.657753   7.498773  10.258607

La moyenne des valeurs des attributs selon femmes et hommes s'ils matchent avec le partenaire:


attr1_1    sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1
Femmes  18.522098  18.463725  21.853816  17.937505  11.879638  11.433490
Hommes  30.344345  14.936068  19.722747  18.195766   7.275580   9.649348

La moyenne des valeurs des attributs selon femmes et hommes s'ils ne matchent pas avec le partenaire:


attr1_1    sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1
Femmes  18.841649  18.359954  21.512021  16.866749  12.030554  12.429795
Hommes  28.866248  16.487874  19.533558  17.552168   7.542446  10.377823

In [92]:
# Visualisation des résultats
# Moyennes des attributs avant le speed date par genre
fig1 = px.bar(df_question11_global, barmode='group', title="Moyennes des attributs avant le speed date par genre")
fig1.update_layout(xaxis_title="Genre", yaxis_title="Moyenne des attributs", legend_title="Attributs")
fig1.show()

# Moyennes des attributs lorsque les participants matchent par genre
fig2 = px.bar(df_question121_global, barmode='group', title="Moyennes des attributs lorsque les participants matchent par genre")
fig2.update_layout(xaxis_title="Genre", yaxis_title="Moyenne des attributs", legend_title="Attributs")
fig2.show()

# Moyennes des attributs lorsque les participants ne matchent pas par genre
fig3 = px.bar(df_question120_global, barmode='group', title="Moyennes des attributs lorsque les participants ne matchent pas par genre")
fig3.update_layout(xaxis_title="Genre", yaxis_title="Moyenne des attributs", legend_title="Attributs")
fig3.show()

Nous remarquons alors que les attributs les moins désirables par les hommes et les femmes avant le speed date, s'ils matchent ou s'ils ne matchent pas sont désormais amb1_1 et shar1_1 ce qui correspond à l'ambition et le partage des centres d'intérêts.
Pour les femmes, l'attribut le plus important est l'intelligence du partenaire.
Pour les hommes, l'attribut le plus important est l'attractivité du partenaire.

QUESTION2: IMPORTANCE DE L'ATTRACTIVITE DANS LA SELECTION DU PARTENAIRE PAR RAPPORT A SON IMPACT REEL

In [93]:
# Le lendemain du speed date & si les femmes/hommes matchent

df_question201_femme = df_femmes_waves_match1['attr7_2']
df_question211_homme = df_hommes_waves_match1['attr7_2']

# Le lendemain du speed date & si les femmes/hommes ne matchent pas

df_question210_homme = df_hommes_waves_match0['attr7_2']
df_question200_femme = df_femmes_waves_match0['attr7_2']

# 3/4 semaines après le speed date si les hommes/femmes matchent

df_question201_femme_apres = df_femmes_waves_match1['attr7_3']
df_question211_homme_apres = df_hommes_waves_match1['attr7_3']

# 3/4 semaines après le speed date si les hommes/femmes ne matchent pas
df_question201_femme_apres = df_femmes_waves_match0['attr7_3']
df_question210_homme_apres = df_hommes_waves_match0['attr7_3']


In [94]:
# Le lendemain du speed date & si les femmes/hommes matchent

importance_att_01_lendemain_mean = df_question201_femme.mean()
importance_att_11_lendemain_mean = df_question211_homme.mean()
data4=np.vstack([importance_att_01_lendemain_mean, importance_att_11_lendemain_mean])
df_global_att_lendemain1=pd.DataFrame(data4.T, columns=['Femmes','Hommes'], index=['attr7_2'])
print("La moyenne de l'attractivité le lendemain du speed date et s'ils matchent:")
display(df_global_att_lendemain1)

# Le lendemain du speed date & si les femmes/hommes ne matchent pas
importance_att_00_lendemain_mean = df_question210_homme.mean()
importance_att_10_lendemain_mean = df_question200_femme.mean()
data5=np.vstack([importance_att_00_lendemain_mean, importance_att_10_lendemain_mean])
df_global_att_lendemain0=pd.DataFrame(data5.T, columns=['Femmes','Hommes'], index=['attr7_2'])
print("La moyenne de l'attractivité le lendemain du speed date et s'ils ne matchent pas:")
display(df_global_att_lendemain0)

# 3/4 semaines après le speed date si les hommes/femmes matchent
importance_att_01_apres_mean= df_question201_femme_apres.mean()
importance_att_11_apres_mean = df_question211_homme_apres.mean()
data6=np.vstack([importance_att_01_apres_mean, importance_att_11_apres_mean])
df_global_att_apres1=pd.DataFrame(data6.T, columns=['Femmes','Hommes'], index=['attr7_3'])
print("La moyenne de l'attractivité 3/4 semaines après speed date et s'ils matchent:")
display(df_global_att_apres1)

# 3/4 semaines après le speed date si les hommes/femmes ne matchent pas
importance_att_00_apres_mean = df_question201_femme_apres.mean()
importance_att_10_apres_mean= df_question210_homme_apres.mean()
data7=np.vstack([importance_att_00_apres_mean, importance_att_10_apres_mean])
df_global_att_apres0=pd.DataFrame(data7.T, columns=['Femmes','Hommes'], index=['attr7_3'])
print("La moyenne de l'attractivité 3/4 semaines après speed date et s'ils ne matchent pas:")
display(df_global_att_apres0)



La moyenne de l'attractivité le lendemain du speed date et s'ils matchent:


Femmes     Hommes
attr7_2  27.273256  37.878981

La moyenne de l'attractivité le lendemain du speed date et s'ils ne matchent pas:


Femmes     Hommes
attr7_2  37.20375  28.904094

La moyenne de l'attractivité 3/4 semaines après speed date et s'ils matchent:


Femmes     Hommes
attr7_3  27.393407  38.427673

La moyenne de l'attractivité 3/4 semaines après speed date et s'ils ne matchent pas:


Femmes     Hommes
attr7_3  27.393407  36.114401

In [95]:
# Transformations réalisées sur nos derniers tableaux pour visualisation des résultats

df_global_att_lendemain1['Moment'] = "Le lendemain"
df_global_att_lendemain1['Match'] = "Oui"

df_global_att_lendemain0['Moment'] = "Le lendemain"
df_global_att_lendemain0['Match'] = "Non"

df_global_att_apres1['Moment'] = "3/4 semaines après"
df_global_att_apres1['Match'] = "Oui"

df_global_att_apres0['Moment'] = "3/4 semaines après"
df_global_att_apres0['Match'] = "Non"

# Concaténation des DataFrames dans un seul pour la visualisation
df_vis = pd.concat([df_global_att_lendemain1, df_global_att_lendemain0, 
                    df_global_att_apres1, df_global_att_apres0], axis=0).reset_index()

# Transformer le DataFrame de large à long
df_vis_long = pd.melt(df_vis, id_vars=['index', 'Moment', 'Match'], var_name='Genre', value_name='Moyenne')



In [96]:
# Visualisation des résultats

fig = px.bar(df_vis_long, 
             x='Moment', 
             y='Moyenne', 
             color='Genre', 
             facet_col='Match', 
             barmode='group',
             title="Moyenne de l'attractivité selon le genre et le scénario",
             labels={'Moment': 'Moment après le speed date', 'Moyenne': 'Moyenne de l\'attractivité', 'Genre': 'Genre'},
             category_orders={"Moment": ["Le lendemain", "3/4 semaines après"], "Match": ["Oui", "Non"]})
fig.show()

Nous constatons d'après les graphes que l'attractivité reste un attribut très important quelque soit la décision des participants et quelque soit le moment après le date.

QUESTION3: LES INTERETS PARTAGES SONT-ILS PLUS IMPORTANTS QU'UNE ORIGINE RACIALE PARTAGEE?

In [97]:
# Importance de l'origine raciale pour la sélection d'un partenaire
    # Pourcentage des personnes qui souhaitent que leur partenaire soit de la même race

df_waves = df[((df['wave'] <= 5) | (df['wave'] >= 10))]
df_waves_imprace_mean_id=df_waves.groupby(df_waves['iid'])['imprace'].mean() 

missing_percentage = df_waves['samerace'].isna().mean() * 100
print("le pourcentage des valeurs manquantes de la colonne samerace est:", missing_percentage)

    # Calculer le nombre de personnes qui souhaitent que leurs partenaires soient de la même race
same_race = df_waves[df_waves['samerace'] == 1]
unique_same_race = same_race.drop_duplicates(subset='iid')
count_unique_people = unique_same_race.shape[0]
print("Le nombre de personnes qui souhaitent que leurs partenaires soient de la même race est:", count_unique_people)
pourcentage_samerace=(count_unique_people/449)*100
print("le pourcentage des personnes qui souhaitent que leurs partenaires soient de la même race:", pourcentage_samerace )
pourcentage_not_samerace = 100 - pourcentage_samerace

# Visualisation des résultats

labels = ['Souhaitent un partenaire de la même race', 'Ne souhaitent pas un partenaire de la même race']
values = [pourcentage_samerace, pourcentage_not_samerace]

fig = px.pie(values=values, names=labels, title='Pourcentage des personnes souhaitant un partenaire de la même race')

fig.show()

le pourcentage des valeurs manquantes de la colonne samerace est: 0.0
Le nombre de personnes qui souhaitent que leurs partenaires soient de la même race est: 411
le pourcentage des personnes qui souhaitent que leurs partenaires soient de la même race: 91.53674832962137


D'après cette analyse globale du dataset, nous constatons que la grande majorité des participants souhaitent que leurs partenaires soient de la même race.

Analysons maintenant plus en détails les données de notre dataset concernant l'origine raciale

In [98]:
# Importance de l'origine raciale pour la sélection d'un partenaire
    #Transformation des valeurs de la colonne imprace

imprace = df_waves_imprace_mean_id.apply(lambda x: "Absolument important(score=10)" if x==10
                                else "Extrêmement important(score=9)" if x ==9
                                else "Très Très important(score=8)" if x==8
                                else "Très important(score=7)" if x ==7
                                else "Important(score=6)" if x==6
                                else "moyen(score=5)" if x==5
                                else "Très peu important(score=4)" if x==4
                                else "Pas assez important(score=3)" if x==3
                                else "Pas important(score=2)" if x==2
                                else "Pas du tout important(score=1)" if x ==1
                                else "Totalement indifférent(score=0)")

In [99]:
# Importance de l'origine raciale pour la sélection d'un partenaire
    # Compter les valeurs de la colonne imprace
imprace_count = imprace.value_counts().sort_values()
imprace_count

imprace
Totalement indifférent(score=0)      7
Absolument important(score=10)      13
Extrêmement important(score=9)      17
Important(score=6)                  26
Très peu important(score=4)         27
Très important(score=7)             27
Très Très important(score=8)        32
moyen(score=5)                      32
Pas important(score=2)              50
Pas assez important(score=3)        56
Pas du tout important(score=1)     162
Name: count, dtype: int64

In [100]:
# Importance de l'origine raciale pour la sélection d'un partenaire
    #Visualisation des résultats - Histogramme

data = {
    'Importance': [
        'Totalement indifférent(score=0)', 'Absolument important(score=10)', 'Extrêmement important(score=9)',
        'Important(score=6)', 'Très peu important(score=4)', 'Très important(score=7)',
        'Très Très important(score=8)', 'moyen(score=5)', 'Pas important(score=2)',
        'Pas assez important(score=3)', 'Pas du tout important(score=1)'
    ],
    'Count': [7, 13, 17, 26, 27, 27, 32, 32, 50, 56, 162]
}

df_visu = pd.DataFrame(data)

fig = px.bar(df_visu, x='Importance', y='Count',
             title="Importance de l'origine raciale partagée dans la sélection d'un partenaire",
             labels={'Count': 'Nombre de Réponses', "Importance": "Niveau d'Importance"},
             color='Count',  
             text='Count')  


fig.update_layout(
    xaxis_title="Niveau d'Importance",
    yaxis_title='Nombre de Réponses',
    xaxis={'categoryorder': 'total descending'},  # Trier par ordre décroissant des réponses
    xaxis_tickangle=-45  # Incliner les labels pour une meilleure lisibilité
)

fig.show()

In [101]:
# Importance de l'origine raciale pour la sélection d'un partenaire
    #Visualisation des résultats - Camembert

fig2 = px.pie(df_visu, names='Importance', values='Count',
             title="Distribution de l'Importance de la Race",
             labels={'Importance': 'Niveau d\'Importance'},
             color_discrete_sequence=px.colors.sequential.RdBu)  # Utilisation d'une séquence de couleurs

# Mise à jour de la mise en page pour améliorer la lisibilité
fig2.update_traces(textposition='inside', textinfo='percent')  # Affichage des pourcentages à l'intérieur


fig2.show()

D'après les formulaires remplis avant l'évènement, nous remarquons que l'origine raciale partagée n'a pas vraiment d'importance pour les participants, ce qui fait que 36.1% des personnes pensent que l'origine raciale partagée n'a pas du tout d'importance pour la sélection d'un partenaire.

In [102]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Avant le speed date
# Calcul du nombre des valeurs manquantes des colonnes ci-dessous

col_list = ['shar1_1','shar','shar1_s','shar1_2','shar7_2', 'shar1_3','shar7_3']
missing_values_count = df_waves[col_list].isnull().sum()
missing_values_df = pd.DataFrame(missing_values_count, columns=['Nombre de valeurs manquantes'])
print(missing_values_df)
print("le nombre de ligne de notre dataset:",df_waves.shape)

         Nombre de valeurs manquantes
shar1_1                           116
shar                              845
shar1_s                          4166
shar1_2                           751
shar7_2                          4842
shar1_3                          3785
shar7_3                          4800
le nombre de ligne de notre dataset: (6816, 195)


Nous n'allons pas prendre en considération les colonnes qui contiennent plus que la moitié du dataset en valeurs manquantes.
Ce sont les colonnes: shar1_s, shar7_2, shar1_3, shar7_3

In [103]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Avant le speed date
# Calcul de la moyenne de la colonne shar1_1 par iid

df_waves_shar11_mean_id=df_waves.groupby(df_waves['iid'])['shar1_1'].mean() 

In [104]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Avant le speed date
# Transformation des valeurs de la colonne shar1_1

shar11 = df_waves_shar11_mean_id.apply(lambda x: "Absolument important(score=100)" if x==100
                                else "Extrêmement important(score=90)" if x ==90
                                else "Très Très important(score=80)" if x==80
                                else "Très important(score=70)" if x ==70
                                else "Important(score=60)" if x==60
                                else "moyen(score=50)" if x==50
                                else "Très peu important(score=40)" if x==40
                                else "Pas assez important(score=30)" if x==30
                                else "Pas important(score=20)" if x==20
                                else "Pas du tout important(score=10)" if x ==10
                                else "Totalement indifférent(score=0)")

In [105]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Avant le speed date
    # Compter les valeurs de la colonne shar1_1
shar11_count = shar11.value_counts().sort_values()
shar11_count

shar1_1
Pas assez important(score=30)        6
Pas important(score=20)             62
Pas du tout important(score=10)    125
Totalement indifférent(score=0)    256
Name: count, dtype: int64

In [106]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Avant le speed date
    # Visualisation des résultats - Camembert

data_shar11 = {
    'Importance': [
        'Totalement indifférent(score=0)', 'Absolument important(score=10)', 'Extrêmement important(score=9)',
        'Important(score=6)', 'Très peu important(score=4)', 'Très important(score=7)',
        'Très Très important(score=8)', 'moyen(score=5)', 'Pas important(score=2)',
        'Pas assez important(score=3)', 'Pas du tout important(score=1)'
    ],
    'Count': [256, 0, 0, 0, 0, 0, 0, 0, 62, 6, 125]
}
df_visu3=pd.DataFrame(data_shar11)
fig2 = px.pie(df_visu3, names='Importance', values='Count',
             title="Distribution de l'Importance des centres d'intérêts partagés",
             labels={'Importance': "Niveau d'Importance"},
             color_discrete_sequence=px.colors.sequential.RdBu)  # Utilisation d'une séquence de couleurs

# Mise à jour de la mise en page pour améliorer la lisibilité
fig2.update_traces(textposition='inside', textinfo='percent')  # Affichage des pourcentages à l'intérieur


fig2.show()

In [107]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Après le speed date
# Calcul de la moyenne de la colonne shar par iid

df_waves_shar1_mean_id=df_waves.groupby(df_waves['iid'])['shar'].mean() #si match=1
df_waves_shar0_mean_id=df_waves.groupby(df_waves['iid'])['shar'].mean() #si match=0

In [108]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Après le speed date
# Transformation des valeurs de la colonne shar

def map_score_to_label(score):
    labels = {
        10: "Absolument important(score=10)",
        9: "Extrêmement important(score=9)",
        8: "Très Très important(score=8)",
        7: "Très important(score=7)",
        6: "Important(score=6)",
        5: "moyen(score=5)",
        4: "Très peu important(score=4)",
        3: "Pas assez important(score=3)",
        2: "Pas important(score=2)",
        1: "Pas du tout important(score=1)",
        0: "Totalement indifférent(score=0)"
    }
    return labels.get(score, "Inconnu")  # Retourne 'Inconnu' si le score n'est pas dans le dictionnaire

data = {'shar1_mean_id': [10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0], #si match=1
        'shar0_mean_id': [9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 10]} #si match=0
df_map = pd.DataFrame(data)

df_map['shar1'] = df_map['shar1_mean_id'].apply(map_score_to_label)
df_map['shar0'] = df_map['shar0_mean_id'].apply(map_score_to_label)


In [109]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Après le speed date
    # Compter les valeurs de la colonne shar1 et shar0
shar1_count = df_map['shar1'].value_counts().sort_values()
print(shar1_count)

shar0_count = df_map['shar0'].value_counts().sort_values()
print(shar0_count)

shar1
Absolument important(score=10)     1
Extrêmement important(score=9)     1
Très Très important(score=8)       1
Très important(score=7)            1
Important(score=6)                 1
moyen(score=5)                     1
Très peu important(score=4)        1
Pas assez important(score=3)       1
Pas important(score=2)             1
Pas du tout important(score=1)     1
Totalement indifférent(score=0)    1
Name: count, dtype: int64
shar0
Extrêmement important(score=9)     1
Très Très important(score=8)       1
Très important(score=7)            1
Important(score=6)                 1
moyen(score=5)                     1
Très peu important(score=4)        1
Pas assez important(score=3)       1
Pas important(score=2)             1
Pas du tout important(score=1)     1
Totalement indifférent(score=0)    1
Absolument important(score=10)     1
Name: count, dtype: int64


In [110]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - le lendemain du speed date
    # Calcul de la moyenne de la colonne shar par iid

df_waves_shar121_mean_id=df_waves.groupby(df_waves['iid'])['shar1_2'].mean() #si match=1
df_waves_shar120_mean_id=df_waves.groupby(df_waves['iid'])['shar1_2'].mean() #si match=0

In [111]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - le lendemain du speed date
# Transformation des valeurs de la colonne shar1_2

def map_score_to_label(score):
    labels = {
        100: "Absolument important(score=100)",
        90: "Extrêmement important(score=90)",
        80: "Très Très important(score=80)",
        70: "Très important(score=70)",
        60: "Important(score=60)",
        50: "moyen(score=50)",
        40: "Très peu important(score=40)",
        30: "Pas assez important(score=30)",
        20: "Pas important(score=20)",
        10: "Pas du tout important(score=10)",
        0: "Totalement indifférent(score=0)"
    }
    return labels.get(score, "Inconnu")  # Retourne 'Inconnu' si le score n'est pas dans le dictionnaire

data = {'shar121_mean_id': [100, 90, 80, 70, 60, 50, 40, 30, 20, 10, 0], #si match=1
        'shar120_mean_id': [100,90, 80, 70, 60, 50, 40, 30, 20, 10, 0]} #si match=0
df_map = pd.DataFrame(data)

df_map['shar121'] = df_map['shar121_mean_id'].apply(map_score_to_label)
df_map['shar120'] = df_map['shar120_mean_id'].apply(map_score_to_label)


In [112]:
# Evaluation de l'importance d'avoir des centres d'intérêts partagés pour la séléction d'un partenaire - Le lendemain du speed date
    # Compter les valeurs de la colonne shar120 et shar120
shar121_count = df_map['shar121'].value_counts().sort_values()
print(shar121_count)

shar120_count = df_map['shar120'].value_counts().sort_values()
print(shar120_count)

shar121
Absolument important(score=100)    1
Extrêmement important(score=90)    1
Très Très important(score=80)      1
Très important(score=70)           1
Important(score=60)                1
moyen(score=50)                    1
Très peu important(score=40)       1
Pas assez important(score=30)      1
Pas important(score=20)            1
Pas du tout important(score=10)    1
Totalement indifférent(score=0)    1
Name: count, dtype: int64
shar120
Absolument important(score=100)    1
Extrêmement important(score=90)    1
Très Très important(score=80)      1
Très important(score=70)           1
Important(score=60)                1
moyen(score=50)                    1
Très peu important(score=40)       1
Pas assez important(score=30)      1
Pas important(score=20)            1
Pas du tout important(score=10)    1
Totalement indifférent(score=0)    1
Name: count, dtype: int64


data_shar121=data_shar120, donc les graphes seront aussi similaires.

Nous constatons après analyse des données avant, après et le lendemain du speed date que le fait d'avoir des centre d'intérêts partagés n'a pas du tout d'importance pour la sélection du partenaire.
Alors, ni le fait d'avoir des centres d'intérêts partagés, ni le fait d'avoir le même origine n'impactent la décision des participants pour sélectionner un partenaire.
la toute première analyse du dataset montre une légère préférence pour l'origine raciale partagée puisque 91% des personnes souhaitent que leur partenaire soit de la même race.

QUESTION4: LES GENS PEUVENT-ILS PREDIRE AVEC PRECISION LEUR PROPRE VALEUR PERCUE SUR LE MARCHE DES RENCONTRE?

In [113]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Calcul du nombre des valeurs manquantes des colonnes ci-dessous

col_list1 = ['expnum','attr3_1','sinc3_1','intel3_1','fun3_1','amb3_1','attr5_1','sinc5_1','intel5_1','fun5_1','amb5_1','like','prob','dec','match_es','attr3_s','sinc3_s','intel3_s','fun3_s','amb3_s','satis_2','attr5_2','sinc5_2','intel5_2','fun5_2','amb5_2','attr3_3','sinc3_3','intel3_3','fun3_3','amb3_3','attr5_3','sinc5_3','intel5_3','fun5_3','amb5_3','attr','sinc','intel','fun','amb']
missing_values_count1 = df_waves[col_list1].isnull().sum()

missing_values_df1 = pd.DataFrame(missing_values_count1, columns=['Nombre de valeurs manquantes'])

print(missing_values_df1)


          Nombre de valeurs manquantes
expnum                            5016
attr3_1                            100
sinc3_1                            100
intel3_1                           100
fun3_1                             100
amb3_1                             100
attr5_1                           1910
sinc5_1                           1910
intel5_1                          1910
fun5_1                            1910
amb5_1                            1910
like                               212
prob                               270
dec                                  0
match_es                           538
attr3_s                           4202
sinc3_s                           4202
intel3_s                          4202
fun3_s                            4202
amb3_s                            4202
satis_2                            751
attr5_2                           2439
sinc5_2                           2439
intel5_2                          2439
fun5_2                   

Nous contatons que les colonnes: expnum, attr5_1, sinc5_1, intel5_1, fun5_1, amb5_1, attr3_s, sinc3_s, intel3_s, fun3_s, amb3_s, attr5_2,sinc5_2,intel5_2, fun5_2, attr3_3, sinc3_3, intel3_3, fun3_3, amb3_3, attr5_3, sinc5_3, intel5_3, fun5_3, amb5_3 contiennent beaucoup de valeurs manquantes, donc nous allons pas les prendre en compte dans notre analyse.

In [114]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Calcul de la moyenne des valeurs des formulaires remplis pendant et avant l'évènement

col_list1 = ['match','gender','attr3_1','sinc3_1','intel3_1','fun3_1','amb3_1']
df_wave_list1=df_waves[col_list1].reset_index(drop=True)
df_list1_h=df_wave_list1[df_wave_list1['gender']==1].reset_index(drop=True)
df_list1_f=df_wave_list1[df_wave_list1['gender']==0].reset_index(drop=True)

df_list1_f1=df_list1_f[df_list1_f['match']==1]
df_list1_f0=df_list1_f[df_list1_f['match']==0]
df_list1_h0=df_list1_h[df_list1_h['match']==0]
df_list1_h1=df_list1_h[df_list1_h['match']==1]

col_list2 = ['match','gender','attr','sinc','intel','fun','amb']
df_wave_list2=df_waves[col_list2].reset_index(drop=True)
df_list2_h=df_wave_list2[df_wave_list2['gender']==1].reset_index(drop=True)
df_list2_f=df_wave_list2[df_wave_list2['gender']==0].reset_index(drop=True)

df_list2_f1=df_list2_f[df_list2_f['match']==1]
df_list2_f0=df_list2_f[df_list2_f['match']==0]
df_list2_h0=df_list2_h[df_list2_h['match']==0]
df_list2_h1=df_list2_h[df_list2_h['match']==1]

moyenne31_femme1=df_list1_f1.mean()
moyenne31_homme1=df_list1_h1.mean()
moyenne31_femme0=df_list1_f0.mean()
moyenne31_homme0=df_list1_h0.mean()

moyenne_femme1=df_list2_f1.mean()
moyenne_homme1=df_list2_h1.mean()
moyenne_femme0=df_list2_f0.mean()
moyenne_homme0=df_list2_h0.mean()

data1 = [
    {'Groupe': 'Femme Match=1', 'Attr3_1': moyenne31_femme1['attr3_1'], 'Sinc3_1': moyenne31_femme1['sinc3_1'], 'Intel3_1': moyenne31_femme1['intel3_1'], 'Fun3_1': moyenne31_femme1['fun3_1'], 'Amb3_1': moyenne31_femme1['amb3_1']},
    {'Groupe': 'Homme Match=1', 'Attr3_1': moyenne31_homme1['attr3_1'], 'Sinc3_1': moyenne31_homme1['sinc3_1'], 'Intel3_1': moyenne31_homme1['intel3_1'], 'Fun3_1': moyenne31_homme1['fun3_1'], 'Amb3_1': moyenne31_homme1['amb3_1']},
    {'Groupe': 'Femme Match=0', 'Attr3_1': moyenne31_femme0['attr3_1'], 'Sinc3_1': moyenne31_femme0['sinc3_1'], 'Intel3_1': moyenne31_femme0['intel3_1'], 'Fun3_1': moyenne31_femme0['fun3_1'], 'Amb3_1': moyenne31_femme0['amb3_1']},
    {'Groupe': 'Homme Match=0', 'Attr3_1': moyenne31_homme0['attr3_1'], 'Sinc3_1': moyenne31_homme0['sinc3_1'], 'Intel3_1': moyenne31_homme0['intel3_1'], 'Fun3_1': moyenne31_homme0['fun3_1'], 'Amb3_1': moyenne31_homme0['amb3_1']}
]
data2 = [
    {'Groupe': 'Femme Match=1', 'Attr': moyenne_femme1['attr'], 'Sinc': moyenne_femme1['sinc'], 'Intel': moyenne_femme1['intel'], 'Fun': moyenne_femme1['fun'], 'Amb': moyenne_femme1['amb']},
    {'Groupe': 'Homme Match=1', 'Attr': moyenne_homme1['attr'], 'Sinc': moyenne_homme1['sinc'], 'Intel': moyenne_homme1['intel'], 'Fun': moyenne_homme1['fun'], 'Amb': moyenne_homme1['amb']},
    {'Groupe': 'Femme Match=0', 'Attr': moyenne_femme0['attr'], 'Sinc': moyenne_femme0['sinc'], 'Intel': moyenne_femme0['intel'], 'Fun': moyenne_femme0['fun'], 'Amb': moyenne_femme0['amb']},
    {'Groupe': 'Homme Match=0', 'Attr': moyenne_homme0['attr'], 'Sinc': moyenne_homme0['sinc'], 'Intel': moyenne_homme0['intel'], 'Fun': moyenne_homme0['fun'], 'Amb': moyenne_homme0['amb']}
]
# Création du DataFrame
df_moyennes1 = pd.DataFrame(data1)
df_moyennes2 = pd.DataFrame(data2)

# Afficher le DataFrame créé
print(df_moyennes1)
print(df_moyennes2)


          Groupe   Attr3_1   Sinc3_1  Intel3_1    Fun3_1    Amb3_1
0  Femme Match=1  7.113346  8.325411  8.146252  7.802559  7.307130
1  Homme Match=1  7.103825  8.136612  8.535519  7.927140  7.757741
2  Femme Match=0  7.068719  8.444166  8.206872  7.798855  7.538654
3  Homme Match=0  6.900212  8.095895  8.457183  7.478061  7.481953
          Groupe      Attr      Sinc     Intel       Fun       Amb
0  Femme Match=1  7.066964  7.749553  8.024150  7.597122  7.365201
1  Homme Match=1  7.536542  7.827338  7.848837  7.603943  7.192737
2  Femme Match=0  5.648560  6.889725  7.251845  5.939164  6.766641
3  Homme Match=0  6.209173  7.137899  7.166422  6.307621  6.479401


In [115]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Visualisation des résultats

# Extraire les données pour les femmes et les hommes avec Match=0
femme_match0 = df_moyennes1[df_moyennes1['Groupe'] == 'Femme Match=0']
homme_match0 = df_moyennes2[df_moyennes2['Groupe'] == 'Homme Match=0']

# Attributs à comparer
attributes_3_1 = ['Attr3_1', 'Sinc3_1', 'Intel3_1', 'Fun3_1', 'Amb3_1']
attributes = ['Attr', 'Sinc', 'Intel', 'Fun', 'Amb']

# Création d'une figure Plotly
fig = go.Figure()

# Ajout des barres pour les femmes avec attributs 3_1
for attr in attributes_3_1:
    fig.add_trace(go.Bar(
        name=f'{attr} - Femmes',
        x=[attr.replace('3_1', '')],  # Simplifier le nom de l'attribut pour la visualisation
        y=femme_match0[attr],
        marker_color='rgba(255, 85, 85, 0.6)'
    ))

# Ajout des barres pour les hommes avec attributs de data2
for attr in attributes:
    fig.add_trace(go.Bar(
        name=f'{attr} - Hommes',
        x=[attr],  # Les noms des attributs pour les hommes sont déjà simplifiés
        y=homme_match0[attr],
        marker_color='rgba(85, 85, 255, 0.6)'
    ))

# Mise à jour de la mise en page
fig.update_layout(
    barmode='group',
    title="Comparaison des Attributs pour Femmes et Hommes sans Match",
    xaxis_title="Attributs",
    yaxis_title="Valeur Moyenne",
    legend_title="Groupes et Attributs"
)

fig.show()

In [116]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Visualisation des résultats

# Extraire les données pour les femmes et les hommes avec Match=1
femme_match1 = df_moyennes1[df_moyennes1['Groupe'] == 'Femme Match=1'].iloc[0]
homme_match1 = df_moyennes2[df_moyennes2['Groupe'] == 'Homme Match=1'].iloc[0]

# Attributs à comparer
attributes_3_1 = ['Attr3_1', 'Sinc3_1', 'Intel3_1', 'Fun3_1', 'Amb3_1']
attributes = ['Attr', 'Sinc', 'Intel', 'Fun', 'Amb']

# Création d'une figure Plotly
fig = go.Figure()

# Ajout des barres pour les femmes avec attributs 3_1
for attr, data_attr in zip(attributes, attributes_3_1):
    fig.add_trace(go.Bar(
        name=f'{data_attr} - Femmes',
        x=[attr],  # Simplifier le nom de l'attribut pour la visualisation
        y=[femme_match1[data_attr]],
        marker_color='rgba(255, 85, 85, 0.6)'
    ))

# Ajout des barres pour les hommes avec attributs de data2
for attr in attributes:
    fig.add_trace(go.Bar(
        name=f'{attr} - Hommes',
        x=[attr],
        y=[homme_match1[attr]],
        marker_color='rgba(85, 85, 255, 0.6)'
    ))

# Mise à jour de la mise en page
fig.update_layout(
    barmode='group',
    title="Comparaison des Attributs pour Femmes et Hommes avec Match=1",
    xaxis_title="Attributs",
    yaxis_title="Valeur Moyenne",
    legend_title="Groupes et Attributs",
    xaxis=dict(tickmode='linear')
)

# Afficher le graphique
fig.show()

In [117]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Visualisation des résultats

# Extraire les données pour les hommes avec Match=0 de data1 et les femmes avec Match=0 de data2
homme_match0 = df_moyennes1[df_moyennes1['Groupe'] == 'Homme Match=0'].iloc[0]
femme_match0 = df_moyennes2[df_moyennes2['Groupe'] == 'Femme Match=0'].iloc[0]

# Attributs à comparer
attributes_3_1 = ['Attr3_1', 'Sinc3_1', 'Intel3_1', 'Fun3_1', 'Amb3_1']
attributes = ['Attr', 'Sinc', 'Intel', 'Fun', 'Amb']

# Création d'une figure Plotly
fig = go.Figure()

# Ajout des barres pour les hommes avec attributs 3_1 de data1
for attr, data_attr in zip(attributes, attributes_3_1):
    fig.add_trace(go.Bar(
        name=f'{data_attr} - Hommes',
        x=[attr.replace('3_1', '')],  # Simplifier le nom de l'attribut pour la visualisation
        y=[homme_match0[data_attr]],
        marker_color='rgba(85, 85, 255, 0.6)'
    ))

# Ajout des barres pour les femmes avec attributs de data2
for attr in attributes:
    fig.add_trace(go.Bar(
        name=f'{attr} - Femmes',
        x=[attr],
        y=[femme_match0[attr]],
        marker_color='rgba(255, 85, 85, 0.6)'
    ))

# Mise à jour de la mise en page
fig.update_layout(
    barmode='group',
    title="Comparaison des Attributs pour Hommes et Femmes sans Match",
    xaxis_title="Attributs",
    yaxis_title="Valeur Moyenne",
    legend_title="Groupes et Attributs",
    xaxis=dict(tickmode='linear')
)

# Afficher le graphique
fig.show()


In [118]:
# La vraie valeur perçue des personnes sur le marché des rencontres
# Visualisation des résultats

# Extraire les données pour les hommes avec Match=1 de data1 et les femmes avec Match=1 de data2
homme_match1 = df_moyennes1[df_moyennes1['Groupe'] == 'Homme Match=1'].iloc[0]
femme_match1 = df_moyennes2[df_moyennes2['Groupe'] == 'Femme Match=1'].iloc[0]

# Attributs à comparer
attributes_3_1 = ['Attr3_1', 'Sinc3_1', 'Intel3_1', 'Fun3_1', 'Amb3_1']
attributes = ['Attr', 'Sinc', 'Intel', 'Fun', 'Amb']

# Création d'une figure Plotly
fig = go.Figure()

# Ajout des barres pour les hommes avec attributs 3_1 de data1
for attr, data_attr in zip(attributes, attributes_3_1):
    fig.add_trace(go.Bar(
        name=f'{data_attr} - Hommes',
        x=[attr.replace('3_1', '')],  # Simplifier le nom de l'attribut pour la visualisation
        y=[homme_match1[data_attr]],
        marker_color='rgba(85, 85, 255, 0.6)'
    ))

# Ajout des barres pour les femmes avec attributs de data2
for attr in attributes:
    fig.add_trace(go.Bar(
        name=f'{attr} - Femmes',
        x=[attr],
        y=[femme_match1[attr]],
        marker_color='rgba(255, 85, 85, 0.6)'
    ))

# Mise à jour de la mise en page
fig.update_layout(
    barmode='group',
    title="Comparaison des Attributs pour Hommes et Femmes avec Match=1",
    xaxis_title="Attributs",
    yaxis_title="Valeur Moyenne",
    legend_title="Groupes et Attributs",
    xaxis=dict(tickmode='linear')
)

# Afficher le graphique
fig.show()

D'après toutes ces visualisations, nous constatons que les participants à cet évènement arrivent plus ou moins à prédire leur propre valeur sur le marché.

QUESTION5: POUR CE QUI EST D'OBTENIR UN DEUXIEME RENDEZ-VOUS, VAUT-IL MIEUX ETRE LE PREMIER SPEED DATE DE LA SOIREE OU LE DERNIER?

In [119]:
# Importance des ordre de date pour la sélection du partenaire

colonnes_a_prendre=['iid','pid','round','order','match']

df_orderspeed=df_waves[colonnes_a_prendre].reset_index(drop=True)

df_orderspeed['order'].groupby(df_orderspeed['order']).count()

match1=df_orderspeed[df_orderspeed['match']==1]

order_count_match1 = match1['order'].value_counts()
display(order_count_match1)


order
1     103
9      83
4      81
6      78
3      77
5      75
7      73
10     70
2      70
8      69
11     47
14     46
13     43
12     40
18     36
15     32
16     23
19     21
17     21
20     20
22     10
21      6
Name: count, dtype: int64

In [120]:
# Visualisation des résultats

fig = px.bar(order_count_match1, 
             x=order_count_match1.index, 
             y=order_count_match1.values, 
             labels={'x': 'Order', 'y': 'Count'},
             title="Nombre d'occurrences pour chaque 'order' lorsque match est 1")


fig.show()

Conclusion:
Après la réalisation de toutes ces analyses, nous retenons que pour séléctionner un partenaire, plusieurs facteurs rentrent en compte:
- L'origine raciale et le fait d'avoir des centre d'intérêts partagés ne sont pas du tout des critères importants pour la sélection des partenaires
- Les hommes cherchent plus de l'attractivité chez la femme qu'autre chose, tandis que les femmes cherchent plus de l'intelligence chez leur partenaire
- L'attractivité reste un facteur important quelque soit la décision des participants et quelque soit le moment après le date
- Les participants à cet évènement arrivent à percevoir plus ou moins leur propre valeur sur le marché des rencontres
- l'odre du passage dans les speed date représente beaucoup d'importance pour obtenir un match.